src/esp/gfx/PBRDrawable.cpp

```diff
@@ -0,0 +1,141 @@
+      // fragment shader
+      .setObjectId(
+          static_cast<RenderCamera&>(camera).useDrawableIds()
+              ? drawableId_
+              : (materialData_->perVertexObjectId ? 0 : node_.getSemanticId()))
+      .setTransformationMatrix(transformationMatrix)  // modelview matrix
+      .setMVPMatrix(camera.projectionMatrix() * transformationMatrix)
+      // because we actually do not have scaling, only rotation, so that the
+      // modelview matrix mv = inv(mv^T) where mv stands for the up-left 3x3
+      // part of the modelview matrix
+      .setNormalMatrix(transformationMatrix.rotationScaling())
+      .bindTextures(
+          materialData_->albedoTexture, materialData_->roughnessTexture,
+          materialData_->metallicTexture, materialData_->normalTexture)
+      .setBaseColor(materialData_->baseColor)
+      .setRoughness(materialData_->roughness)
+      .setMetallic(materialData_->metallic);
+
+  if (materialData_->textureMatrix != Mn::Matrix3{}) {
+    // TODO
+  }
+
+  shader_->draw(mesh_);
+}
+
+Mn::ResourceKey PBRDrawable::getShaderKey(Mn::UnsignedInt lightCount,
+                                          PBRShader::Flags flags) const {
+  return Corrade::Utility::formatString(
+      SHADER_KEY_TEMPLATE, lightCount,
+      static_cast<PBRShader::Flags::UnderlyingType>(flags));
+}
+
+PBRDrawable& PBRDrawable::updateShader() {
+  PBRShader::Flags flags = PBRShader::Flag::ObjectId;
+
+  if (materialData_->textureMatrix != Mn::Matrix3{}) {
+    // TODO: may be supported in the future
+  }
+  if (materialData_->albedoTexture)
+    flags |= PBRShader::Flag::AlbedoTexture;
+  if (materialData_->roughnessTexture)
+    flags |= PBRShader::Flag::RoughnessTexture;
+  if (materialData_->metallicTexture)
+    flags |= PBRShader::Flag::MetallicTexture;
+  if (materialData_->normalTexture)
+    flags |= PBRShader::Flag::NormalTexture;
+  if (materialData_->perVertexObjectId) {
+    // TODO: may be supported in the future
+  }
+
+  unsigned int lightCount = lightSetup_->size();
+  if (!shader_ || shader_->lightCount() != lightCount ||
+      shader_->flags() != flags) {
+    // if the number of lights or flags have changed, we need to fetch a
+    // compatible shader
+    shader_ = shaderManager_.get<Mn::GL::AbstractShaderProgram, PBRShader>(
+        getShaderKey(lightCount, flags));
+
+    // if no shader with desired number of lights and flags exists, create one
+    if (!shader_) {
+      shaderManager_.set<Mn::GL::AbstractShaderProgram>(
+          shader_.key(), new PBRShader{flags, lightCount},
+          Mn::ResourceDataState::Final, Mn::ResourcePolicy::ReferenceCounted);
+    }
+
+    CORRADE_INTERNAL_ASSERT(shader_ && shader_->lightCount() == lightCount &&
+                            shader_->flags() == flags);
+  }
+
+  return *this;
+}
+
+// update every light's color, intensity, range etc.
+PBRDrawable& PBRDrawable::updateShaderLightParameters() {
+  constexpr float dummyRange = 10000.0;
```

Comment:
[Mn::Constants::inf()](https://doc.magnum.graphics/magnum/structMagnum_1_1Math_1_1Constants.html#a33437fd32195685dc87c2b8b44176c62) is what you want to use here.

I unfortunately overlooked that when it was originally commited by @eundersander, [can you fix that there as well](https://github.com/facebookresearch/habitat-sim/blob/364b2b793818efb588e89463487708d38ae08d8a/src/esp/gfx/GenericDrawable.cpp#L52)? Thanks :pray: